pkg/ingester/otlp/convert.go

```diff
@@ -48,49 +57,44 @@ func ConvertOtelToGoogle(src *otelProfile.Profile) *googleProfile.Profile {
 		gf.Id = uint64(i + 1)
 		dst.Function = append(dst.Function, gf)
 	}
+	funcmap := map[string]uint64{}
+	addfunc := func(s string) uint64 {
+		if _, ok := funcmap[s]; !ok {
+			funcmap[s] = uint64(len(dst.Function)) + 1
+			dst.Function = append(dst.Function, &googleProfile.Function{
+				Id:   funcmap[s],
+				Name: addstr(s),
+			})
+		}
+		return funcmap[s]
+	}
 
-	functionOffset := uint64(len(dst.Function)) + 1
 	dst.Location = []*googleProfile.Location{}
-	locationMappingIndexAddressMap := make(map[uint64]uint64)
 	// Convert locations and mappings
 	for i, loc := range src.Location {
 		gl := convertLocationBack(loc)
 		gl.Id = uint64(i + 1)
 		if len(gl.Line) == 0 {
```

Comment:
So my understanding is that we only create those "placeholder" functions currently, as our query path relies on them to deduplicate. 

I think a comment would be nice explaining that.

Maybe something worth considering doing later in the write path. Because if I sent currently Google pprofs without symbols I have the exact same problem. Or am I missing something 😂

Reply:
Hmm... I didn't test it, but it should be the same if we happen to receive pprof without symbols (lines) 😅

Reply:
I think frontend or read path just skips these locations.

![image](https://github.com/user-attachments/assets/39cef545-4586-4c09-a32f-83759768908e)

vs 

![image](https://github.com/user-attachments/assets/2577738a-28cd-418c-ade2-949bab2447d6)

I think it is mostly needed to show the correct shape of the flamegraph - display that there were some frames we just could not name them